terraform/modules.tf.json.template.py

```diff
@@ -8,12 +9,21 @@
         "chalice_indexer": {
             "source": "./indexer",
             "role_arn": "${aws_iam_role.indexer.arn}",
-            "layer_arn": "${aws_lambda_layer_version.dependencies.arn}"
+            "layer_arn": "${aws_lambda_layer_version.dependencies.arn}",
+            "azul_es_domain":
+                "${data.aws_elasticsearch_domain.azul_es_domain}"
+                if config.share_es_domain else
+                "${aws_elasticsearch_domain.elasticsearch[0]}"
+
         },
         "chalice_service": {
             "source": "./service",
             "role_arn": "${aws_iam_role.service.arn}",
-            "layer_arn": "${aws_lambda_layer_version.dependencies.arn}"
+            "layer_arn": "${aws_lambda_layer_version.dependencies.arn}",
+            "azul_es_domain":
+                "${data.aws_elasticsearch_domain.azul_es_domain}"
+                if config.share_es_domain else
+                "${aws_elasticsearch_domain.elasticsearch[0]}"
```

Comment:
Since you have added code, the fact that these two dicts are more or less duplicates of each other becomes more evident. Can you try to eliminate the duplication either via a list comprehension over ['service', 'indexer'] or by extracting the common dict entries into a separate dict which is ** into each of the concrete dicts.